java/flight/flight-sql/src/main/java/org/apache/arrow/flight/sql/FlightSqlUtils.java

```diff
@@ -0,0 +1,172 @@
+          "Closes a reusable prepared statement resource on the server. \n" +
+                  "Request Message: ActionClosePreparedStatementRequest\n" +
+                  "Response Message: N/A");
+
+  public static final List<ActionType> FLIGHT_SQL_ACTIONS = ImmutableList.of(
+          FLIGHT_SQL_CREATEPREPAREDSTATEMENT,
+          FLIGHT_SQL_CLOSEPREPAREDSTATEMENT
+  );
+
+  /**
+   * Converts {@link java.sql.Types} values returned from JDBC Apis to Arrow types.
+   *
+   * @param jdbcDataType {@link java.sql.Types} value.
+   * @param precision    Precision of the type.
+   * @param scale        Scale of the type.
+   * @return The Arrow equivalent type.
+   */
+  public static ArrowType getArrowTypeFromJDBCType(int jdbcDataType, int precision, int scale) {
+
+    switch (jdbcDataType) {
+      case Types.BIT:
+      case Types.BOOLEAN:
+        return ArrowType.Bool.INSTANCE;
+      case Types.TINYINT:
+        return new ArrowType.Int(BIT_WIDTH8, IS_SIGNED_TRUE);
+      case Types.SMALLINT:
+        return new ArrowType.Int(BIT_WIDTH_16, IS_SIGNED_TRUE);
+      case Types.INTEGER:
+        return new ArrowType.Int(BIT_WIDTH_32, IS_SIGNED_TRUE);
+      case Types.BIGINT:
+        return new ArrowType.Int(BIT_WIDTH_64, IS_SIGNED_TRUE);
+      case Types.FLOAT:
+      case Types.REAL:
+        return new ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE);
+      case Types.DOUBLE:
+        return new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE);
+      case Types.NUMERIC:
+      case Types.DECIMAL:
+        return new ArrowType.Decimal(precision, scale);
+      case Types.DATE:
+        return new ArrowType.Date(DateUnit.DAY);
+      case Types.TIME:
+        return new ArrowType.Time(TimeUnit.MILLISECOND, BIT_WIDTH_32);
+      case Types.TIMESTAMP:
+        return new ArrowType.Timestamp(TimeUnit.MILLISECOND, null);
+      case Types.BINARY:
+      case Types.VARBINARY:
+      case Types.LONGVARBINARY:
+        return ArrowType.Binary.INSTANCE;
+      case Types.NULL:
+        return ArrowType.Null.INSTANCE;
+
+      case Types.CHAR:
+      case Types.VARCHAR:
+      case Types.LONGVARCHAR:
+      case Types.CLOB:
+      case Types.NCHAR:
+      case Types.NVARCHAR:
+      case Types.LONGNVARCHAR:
+      case Types.NCLOB:
+
+      case Types.OTHER:
+      case Types.JAVA_OBJECT:
+      case Types.DISTINCT:
+      case Types.STRUCT:
+      case Types.ARRAY:
+      case Types.BLOB:
+      case Types.REF:
+      case Types.DATALINK:
+      case Types.ROWID:
+      case Types.SQLXML:
+      case Types.REF_CURSOR:
+      case Types.TIME_WITH_TIMEZONE:
+      case Types.TIMESTAMP_WITH_TIMEZONE:
+      default:
+        return ArrowType.Utf8.INSTANCE;
+    }
+  }
+
+  /**
+   * Helper to parse {@link com.google.protobuf.Any} objects to the specific protobuf object.
+   *
+   * @param source the raw bytes source value.
+   * @return the materialized protobuf object.
+   */
+  public static Any parseOrThrow(byte[] source) {
```

Comment:
I think these helpers should take in optional ServerStreamListeners.

Reply:
I don't think so, taking into account the other comments about the callers already handling this.